build/base.mk

```diff
@@ -1,7 +1,9 @@
 UNAME_OS := $(shell uname -s)
 UNAME_ARCH := $(shell uname -m)
 
-CACHE_BASE := $(HOME)/.yarpc-go
+XDG_CACHE_HOME ?= $(HOME)/.cache
+
+CACHE_BASE := $(XDG_CACHE_HOME)/yarpc-go
```

Comment:
XDG_CACHE_HOME is not linux specific, but freedesktop. Which includes most of not all linux distribution.

On MacOSX, from Apple documentation. https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html:

> Put data cache files in the `Library/Caches/` directory. Cache data can be used for any data that needs to persist longer than temporary data, but not as long as a support file. Generally speaking, the application does not require cache data to operate properly, but it can use cache data to improve performance. Examples of cache data include (but are not limited to) database cache files and transient, downloadable content. Note that the system may delete the Caches/ directory to free up disk space, so your app must be able to re-create or download these files as needed.

We should determine the OS, if Mac, use `Library/Caches/`. Else, use `XDG_CACHE_HOME`, else use a generic `~/.cache`.

I couldn't find the equivalent document for MS Windows. But our makefiles are GNU make anyway.

Let me know what you think.